modules/tribes/client/components/JoinButton.component.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import LeaveTribeModal from './LeaveTribeModal';
+import { OverlayTrigger, Tooltip } from 'react-bootstrap';
+
+import * as api from '../api/tribes.api';
+
+function JoinButtonPresentational({ isMember, isLoading, joinLabel='Join', joinedLabel='Joined', tribe, isLogged, onToggle }) {
+  const { t } = useTranslation('tribes');
+
+  const ariaLabel = (isMember) ? t('Leave Tribe') : t(`${joinLabel} ({{label}})`, { label: tribe.label });
+  const title = (isMember) ? t(joinedLabel) : t(joinLabel);
+
+  if (!isLogged) {
+    return <a
+      href={`/signup?tribe=${tribe.slug}`}
+      type="button"
+      className="btn btn-sm btn-default tribe-join"
+    >
+      <i className="icon-plus" /> {title}
+    </a>;
+  }
+
+  const leaveTooltip = <Tooltip id={`tribe-${tribe._id}`} placement="bottom">{t('Leave Tribe')}</Tooltip>;
+  const btn = <button
+    type="button"
+    className={`${isMember ? 'btn-active' : ''} btn btn-sm btn-default tribe-join`}
+    onClick={onToggle}
+    disabled={isLoading}
+    aria-label={ariaLabel}
+  >
+    <i className={(isMember) ? 'icon-ok' : 'icon-plus'} /> {title}
+  </button>;
+
+  if (isMember) {
+    return <OverlayTrigger placement="bottom" overlay={leaveTooltip}>{btn}</OverlayTrigger>;
+  } else {
+    return btn;
+  }
+}
+
+JoinButtonPresentational.propTypes = {
+  isMember: PropTypes.bool.isRequired,
+  isLoading: PropTypes.bool,
+  isLogged: PropTypes.bool.isRequired,
+  joinLabel: PropTypes.string,
+  joinedLabel: PropTypes.string,
+  tribe: PropTypes.object.isRequired,
+  onToggle: PropTypes.func
+};
+
+// @TODO this can (and should) be replaced by other container, when we finish the migration; when we start using redux etc.
+export default function JoinButton({ tribe, user, onUpdated, ...rest }) {
+  // isLeaving controls whether the tribe delete modal is shown
+  const [isLeaving, setIsLeaving] = useState(false);
+
+  const [isUpdating, setIsUpdating] = useState(false);
+
+  const isSelfMember = user && user.memberIds && user.memberIds.indexOf(tribe._id) > -1;
+  const [isMember, setIsMember] = useState(isSelfMember);
```

Comment:
I was wondering what the _self_ meant in `isSelfMember` I guess just to distinguish it from `isMember`? If so
`isMemberInitial` seems clearer to me, as the value passed into `useState()` is the initial value, and on subsequent executions of the JoinButton function, it will use stored state.

Reply:
That's indeed the case. Will rename!

Reply:
Done in c5c33f984a7cdb75721ceed50cd222bc2b55b327